packages/devtools_app/test/inspector/diagnostics_test.dart

```diff
@@ -109,5 +110,39 @@ void main() {
         expect(hoverCardTooltip.enabled(), false);
       });
     });
+
+    testWidgets('approximateNodeWidth', (WidgetTester tester) async {
+      final nodeJson = <String, Object?>{
+        'widgetRuntimeType': 'Row',
+        'renderObject': renderObjectJson,
+        'hasChildren': false,
+        'children': [],
+        'showName': true,
+        'name': 'THE NAME',
+      };
+      final diagnosticWithoutService = RemoteDiagnosticsNode(
+        nodeJson,
+        null,
+        false,
+        null,
+      );
+      final diagnosticsNodeDescription = DiagnosticsNodeDescription(
+        diagnosticWithoutService,
+        debuggerController: MockDebuggerController(),
+      );
+      await tester.pumpWidget(wrap(diagnosticsNodeDescription));
+
+      final aproximateWidth = DiagnosticsNodeDescription.approximateNodeWidth(
+        diagnosticWithoutService,
+      );
+      final nodeSize = tester.getSize(find.descendant(
+        of: find.byType(MouseRegion),
+        matching: find.byType(
+          RichText,
+        ),
+      ));
+      debugDumpApp();
+      expect(nodeSize.width, moreOrLessEquals(aproximateWidth, epsilon: 10.0));
```

Comment:
why do you need an epsilon? Can you fix the calculation so it is exactly right (at least until the icon size changes or something?

Reply:
We need an epsilon here because there isn't really a way for us to get this measurement, without changing the Remote Diagnostic Node widget builder.
The code the tries to find the width of the visible text on a node is an approximation since we don't have access to the widget at the time of checking for visibility.
The focus code and the build code could definitely be changed, but "close enough" seemed like a quick and easy way to get value for our users. :)

Reply:
ok I'm fine with having this epsilon.